# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsContentUtils.h
# Commit: 258c4da663ca
# Full Hash: 258c4da663ca41039139ef94be6ec7e7f3f2333d
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2019-06-10 21:47:35
# Regressor Bug: 1529884
# File Overlap Count: 2
# Description:
#   Bug 1529884 - part 5: Make Document::ExecCommand() use EditorCommand directly as far as possible r=smaug
#   
#   Most commands are dispatched only when the `document` has `contenteditable` or
#   in `designMode`.  In such case, command context is considered with the following
#   order:
# ==============================================================================

diff -r 7af1b30df51c -r 258c4da663ca dom/base/nsContentUtils.h
--- a/dom/base/nsContentUtils.h	Mon Jun 10 10:25:31 2019 +0000
+++ b/dom/base/nsContentUtils.h	Mon Jun 10 10:26:04 2019 +0000
@@ -2651,6 +2651,16 @@
   static mozilla::HTMLEditor* GetHTMLEditor(nsPresContext* aPresContext);
 
   /**
+   * Returns pointer to a text editor if <input> or <textarea> element is
+   * active element in the document for aPresContext, or pointer to HTML
+   * editor if there is (i.e., even if non-editable element has focus or
+   * nobody has focus).  The reason is, HTML editor may handle some input
+   * even if there is no active editing host.
+   * Note that this does not return editor in descendant documents.
+   */
+  static mozilla::TextEditor* GetActiveEditor(nsPresContext* aPresContext);
+
+  /**
    * Returns a LogModule that dump calls from content script are logged to.
    * This can be enabled with the 'Dump' module, and is useful for synchronizing
    * content JS to other logging modules.
